frontend/src/scenes/persons/personsLogic.tsx

```diff
@@ -286,6 +286,24 @@ export const personsLogic = kea<personsLogicType>([
             (featureFlags) => featureFlags[FEATURE_FLAGS.CS_DASHBOARDS],
         ],
         feedEnabled: [(s) => [s.featureFlags], (featureFlags) => !!featureFlags[FEATURE_FLAGS.PERSON_FEED_CANVAS]],
+        primaryDistinctId: [
+            (s) => [s.person],
+            (person): string | null => {
+                // We do not track which distinct ID was created through identify, but we can try to guess
+                const nonUuidDistinctIds = person?.distinct_ids.filter((id) => id?.split('-').length !== 5)
```

Comment:
I went for the `-` segment count instead of a UUID regex because the actual user request included some UUID-adjacent identifiers which were longer than 36 characters: https://posthog.slack.com/archives/C07Q2U4BH4L/p1733826806892899

Figured I'd keep this flexible 🤷

Reply:
yeah, this is a pretty good way of doing this actually – I haven't seen any cases where the non-uuid distinct_id _isn't_ the user-created `distinct_id`, and defaulting to the first one in any other cases seems sane enough.  [Looks like we do this on the backend, too](https://github.com/PostHog/posthog/blob/guess-primary-distinct-id/posthog/api/person.py#L151-L152).  I'm fine with this approach, although it would be nice if we could be even more confident, even if this is just a quality-of-life thing for users.

Not worth blocking on nor spending more cycles on though, IMO.